front/app/components/PostShowComponents/Comments/CommentSorting.tsx

```diff
@@ -1,55 +1,61 @@
-import React, { PureComponent } from 'react';
+import React from 'react';
 import FilterSelector from 'components/FilterSelector';
-import { FormattedMessage } from 'utils/cl-intl';
+import { FormattedMessage, MessageDescriptor } from 'utils/cl-intl';
 import messages from './messages';
-import styled from 'styled-components';
 import { CommentsSort } from 'api/comments/types';
-
-const Container = styled.div``;
+import { Box } from '@citizenlab/cl2-component-library';
 
 interface Props {
-  id?: string | undefined;
   onChange: (value: CommentsSort) => void;
-  selectedValue: CommentsSort[];
+  selectedCommentSort: CommentsSort;
   className?: string;
 }
 
-interface State {}
-
-export default class CommentSorting extends PureComponent<Props, State> {
-  handleOnChange = (selectedValue: [CommentsSort]) => {
-    this.setState({ selectedValue });
-    this.props.onChange(selectedValue[0]);
+const CommentSorting = ({
+  onChange,
+  selectedCommentSort,
+  className,
+}: Props) => {
+  const handleOnChange = (selectedValue: ['new'] | ['-upvotes_count']) => {
+    onChange(selectedValue[0]);
   };
 
-  title = (<FormattedMessage {...messages.commentsSortTitle} />);
-
-  sortOptions = [
-    { text: <FormattedMessage {...messages.mostRecent} />, value: 'new' },
-    {
-      text: <FormattedMessage {...messages.mostUpvoted} />,
-      value: '-upvotes_count',
-    },
-  ];
+  const sortOptionsMessages: { [key in CommentsSort]: MessageDescriptor } = {
+    // '-new' = least recent (date posted, descending)
+    // is the default value we get from the parent
+    '-new': messages.leastRecent,
+    // 'new' = most recent (date posted, ascending)
+    new: messages.mostRecent,
+    // '-upvotes_count' = most votes (votes, descending)
+    '-upvotes_count': messages.mostUpvoted,
+  };
 
-  render() {
-    const { className, selectedValue } = this.props;
+  const getSortOptions = () => {
+    const sortOptions: CommentsSort[] = ['-new', 'new', '-upvotes_count'];
+    return sortOptions.map((sortOption) => {
+      return {
+        text: <FormattedMessage {...sortOptionsMessages[sortOption]} />,
+        value: sortOption,
+      };
+    });
+  };
```

Comment:
Not sure if the best code, but at least it would have caught the missing message for '-new', which was the bug.